subsys/bluetooth/host/gatt.c

```diff
@@ -2542,20 +2583,146 @@ int bt_gatt_notify_cb(struct bt_conn *conn,
 int bt_gatt_notify_multiple(struct bt_conn *conn, uint16_t num_params,
 			    struct bt_gatt_notify_params *params)
 {
-	int i, ret;
+	int i;
+	int err;
+	size_t total_len = 0;
+	struct net_buf *buf;
+	struct notify_data data;
 
 	__ASSERT(params, "invalid parameters\n");
 	__ASSERT(num_params, "invalid parameters\n");
 	__ASSERT(params->attr, "invalid parameters\n");
 
+	if (num_params == 1) {
```

Comment:
Consider using `CHECKIF` for static value checks like this

Reply:
didn't know about that, thanks. I hope no one enables `CONFIG_NO_RUNTIME_CHECKS`.

Reply:
That should be fine if they do, if they _know_ that it never fails. IMO `CHECKIF` should only be used for static checks, such as value ranges, NULL checks, etc. that the caller can know for sure beforehand is correct.